modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/js/Attachment/AttachmentLocalizedObjectField.tsx

```diff
@@ -75,72 +54,45 @@ export default function AttachmentLocalizedObjectField({
 	) => {
 		const newValue = {
 			...value,
-			[currentEditingLocale.localeId]: fileId,
+			[editingLanguageId]: fileId,
 		};
 
 		onChange({target: {value: newValue}});
 
 		setAttachment((previous) => {
 			return {
 				...previous,
-				[currentEditingLocale.localeId]: attachmentValue,
+				[editingLanguageId]: attachmentValue,
 			};
 		});
 	};
 
 	const handleDelete = () => {
-		const currentLocaleId = currentEditingLocale.localeId;
-
-		if (Object.hasOwn(attachment, currentLocaleId)) {
-			const newAttachment = {...attachment};
-			delete newAttachment[currentLocaleId];
-			setAttachment(newAttachment);
+		if (Object.hasOwn(attachment, editingLanguageId)) {
+			setAttachment({
+				...attachment,
+				[editingLanguageId]: undefined,
```

Comment:
I'd rather assign it to `null`, because later on execussion, if we find this as `null`, we know we set it. If we see it as `undefined`, it may be set by JS itself, which automatically returns undefined when it cannot find a property. This is just a suggestion, as there are many assignments to `undefined` in portal, but it is better to assign no to `null` IMHO. Both coerce to false. Later on, check this [thread on stackoverflow](https://stackoverflow.com/questions/6604749/what-reason-is-there-to-use-null-instead-of-undefined-in-javascript).

Reply:
The default behavior of fields is to add undefined when we are replicating the undefined values ​​of the default language ID.

So here we are just keeping the same pattern, if we add null we will be creating a special value just for this case.

Reply:
Yeah, I know. If think this is not the best practice, but let's keep it this way for consistency.